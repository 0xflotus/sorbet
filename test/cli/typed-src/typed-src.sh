#!/bin/bash

main/sorbet --silence-dev-message --typed-source=a.rb test/cli/typed-src/*.rb
