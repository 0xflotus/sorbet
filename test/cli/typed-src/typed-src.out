class A
  sig(x: T.any(String, Integer), y: Integer).returns(Integer)
  def a_method(x, y)
    if x.is_a?(String)
       # <constant:String>$8: <Class:String>
       #
       # <selfMethodTemp>$3: A
       #
       # x: String | Integer
       #
       # y: Integer
       #
       # <ifTemp>$5: TrueClass | FalseClass
       #   Being truthy entails:
       #     x to be String
       #   Being falsy entails:
       #     x NOT to be String
          # x: String
          #
          # y: Integer
      x = x.to_i
          # x: Integer
          #
          # y: Integer
          #
          # <statTemp>$9: String
    end
    # x: Integer
    #
    # y: Integer
    x + y
    # dead=1
    # x: Integer
    #
    # y: Integer
    #
    # <returnMethodTemp>$2: Integer
    #
    # <finalReturn>: <bottom>
  end
end
